xds/credentials/credentials.go

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package credentials supports registering and retreiving credentials for xDS
+// communications.
+//
+// Notice: This package is EXPERIMENTAL and may be changed or removed in a later
+// release.
+package credentials
+
+import (
+	"encoding/json"
+
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/google"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var (
+	// registry is a map from credential type name to Credential builder.
+	registry = make(map[string]Builder)
+)
+
+// Builder interface encapsulates a credential provider
+// that can be used for communicating with the xDS Management server.
+type Builder interface {
+
+	// BuildCredsBundle() returns a credential bundle associated with the Builder.
+	BuildCredsBundle(config json.RawMessage) credentials.Bundle
+}
+
+// GoogleDefaultCredsBuilder encapsulates a Google Default credential to satisfy the Builder interface.
+type GoogleDefaultCredsBuilder struct{}
```

Comment:
IMO this (and `insecure`) should be moved to the google creds package and registered via `init`s there.  Then the `xdsclient/bootstrap` package should blank import them since they are credentials we support by default.

Reply:
Done